src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```
@@ -17,7 +17,111 @@ namespace System.Diagnostics.Tests
+            Assert.Equal(1, p.Id);
+        }
+
+        [Fact]
+        public void TestUseShellExecute_Unix_Succeeds()
+        {
+            using (var p = Process.Start(new ProcessStartInfo { UseShellExecute = true, FileName = "exit", Arguments = "42" }))
+            {
+                Assert.True(p.WaitForExit(WaitInMS));
+                Assert.Equal(42, p.ExitCode);
+            }
+        }
+
+        [Fact]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestPriorityClassUnix()
+        {
+            CreateDefaultProcess();
+
+            ProcessPriorityClass priorityClass = _process.PriorityClass;
+            try
+            {
+                _process.PriorityClass = ProcessPriorityClass.High;
+                Assert.Equal(_process.PriorityClass, ProcessPriorityClass.High);
+
+                _process.PriorityClass = ProcessPriorityClass.Normal;
+                Assert.Equal(_process.PriorityClass, ProcessPriorityClass.Normal);
+            }
+            catch (Win32Exception ex)
+            {
+                Assert.True(!PlatformDetection.IsSuperUser, $"Failed even though superuser {ex.ToString()}");
+            }
+            finally
+            {
+                _process.PriorityClass = priorityClass;
+            }
+        }
+
+        [Fact]
+        [Trait(XunitConstants.Category, XunitConstants.RequiresElevation)]
+        public void TestBasePriorityOnUnix()
+        {
+            CreateDefaultProcess();
+
+            ProcessPriorityClass originalPriority = _process.PriorityClass;
+            Assert.Equal(ProcessPriorityClass.Normal, originalPriority);
+
+            try
+            {
+                SetAndCheckBasePriority(ProcessPriorityClass.High, -11);
+                SetAndCheckBasePriority(ProcessPriorityClass.Idle, 19);
+                SetAndCheckBasePriority(ProcessPriorityClass.Normal, 0);
+            }
+            catch (Win32Exception ex)
```

Rather than eating the failures, can we:
- use ConditionalFact/Theory to skip the tests if they require elevated privileges and those aren't available
- split any tests that mostly work fine without elevated privileges from the parts that do require it
?